### PR TITLE
feat: enhance formatHours to optionally include minutes

### PR DESCRIPTION
**Pull request by pepicrft**
## Summary
- Enhanced the `formatHours` function to optionally display minutes in addition to hours
- Updated Chart.js tooltip formatter to use the new feature for better time precision
- Added comprehensive test coverage for the new functionality

## Changes
- Added `includeMinutes` option to `formatHours` function that formats time as "Xh Ym" when enabled
- Modified Chart.js to pass `{ includeMinutes: true }` to the formatHours tooltip formatter
- Added extensive tests covering various scenarios including edge cases and negative values

## Test plan
- [x] Run existing tests to ensure backward compatibility
- [x] Added new tests for the includeMinutes option
- [x] Tested with various hour values including decimals, whole numbers, and edge cases
- [x] Verified Chart.js tooltip formatting works correctly with the new option

### CODE REVIEW

File: web/js/formatters.js [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+ * @returns {string} Formatted time string (e.g., "1h", "1h 30m", "25h", "168h")
  */
-export function formatHours(hours) {
-  const wholeHours = Math.round(hours);
-  return `${wholeHours}h`;
+export function formatHours(hours, options = {}) {
+  const { includeMinutes = false } = options;
+
+  if (!includeMinutes) {
+    const wholeHours = Math.round(hours);
```
[the remark this answers is not in the document]
Comment (pepicrft) [reply web/js/formatters.js:12]: We do a server-side check to set the right formatter based on that.

**Review by fortmarek — APPROVED** [review]

**Review by cschmatzler — APPROVED** [review]
